cmd/callVariants/callVariants.go

```diff
@@ -18,133 +19,130 @@ import (
+	header.Text = append(header.Text, "##FORMAT=<ID=DP,Number=1,Type=Integer,Description=\"Total Depth\">")
+	header.Text = append(header.Text, "##FORMAT=<ID=AD,Number=R,Type=Integer,Description=\"Depth of Each Allele\">")
+	header.Text = append(header.Text, "##FORMAT=<ID=PV,Number=A,Type=Floatg,Description=\"p value for Each Alternate Allele\">")
+	header.Text = append(header.Text, fmt.Sprintf("#CHROM\tPOS\tID\tREF\tALT\tQUAL\tFILTER\tINFO\tFORMAT\t%s", strings.Join(sampleNames, "\t")))
+	return header
+}
 
-	if samFilesPresent && girafFilesPresent {
-		log.Fatalln("ERROR: Input directories contain both giraf and sam files")
-	}
-	if len(alleleChans) < 2 {
-		log.Fatalln("ERROR: Must input at least two samples (between experimental and normal) to facilitate comparisons. Only", len(alleleChans), "sample was submitted")
-	}
+// inputFiles is a custom type that gets filled by flag.Parse()
+type inputFiles []string
 
-	syncedAllelesChan := alleles.SyncAlleleStreams(ref, memBufferSize, alleleChans...)
+// String to satisfy flag.Value interface
+func (i *inputFiles) String() string {
+	return strings.Join(*i, " ")
+}
 
-	return syncedAllelesChan, normalIDs
+// Set to satisfy flag.Value interface
+func (i *inputFiles) Set(value string) error {
+	*i = append(*i, value)
+	return nil
 }
 
 func main() {
-	var outFile *string = flag.String("out", "", "Write output to a file [.vcf].")
-	var sigThreshold *float64 = flag.Float64("p", 0.05, "Do not output variants with p value greater than this value.")
-	var afThreshold *float64 = flag.Float64("af", 0.01, "Variants with allele frequency less than this value will be treated as p = 1.")
-	var linearReference *string = flag.String("lr", "", "Linear reference used for alignment [.fasta].")
-	var graphReference *string = flag.String("gr", "", "Graph reference used for alignment [.gg].")
-	var experimentalSamples *string = flag.String("i", "", "Input experimental sample(s) [.sam, .giraf, .txt]. Can be a file or a txt file with a list (must have .txt extension) of sample paths.")
-	var normalSamples *string = flag.String("n", "", "Input normal sample(s) [.sam, .giraf, .txt]. Can be a file or a txt file with a list (must have .txt extension) of sample paths. If no normal samples are given, each experimental sample will me measured against the other experimental samples.")
-	var minMapQ *int64 = flag.Int64("minMapQ", 20, "Exclude all reads with mapping quality less than this value")
-	var memBufferSize *int = flag.Int("memBuffer", 100, "Maximum number of allele records to store in memory at once")
-	var minCov *int = flag.Int("minCoverage", 10, "Minimum number of covering reads to be considered a valid variant")
-	flag.Usage = usage
-	log.SetFlags(log.Ldate | log.Ltime | log.Lshortfile)
+	var experimentalFiles, normalFiles inputFiles
+	flag.Var(&experimentalFiles, "i", "Input experimental files. May be declared more than once (.bam, .sam)")
```

Comment:
Just to check, does this "may be declared more than once" mean that the program can handle a combination of both .bam and .sam files?

Reply:
It can handle a combination of files types, but the main purpose is for joint variant calling for multiple samples. It will process them at the same time and use the other samples to determine p values.